### PR TITLE
bump docker dependencies version

### PR DESCRIPTION
**Pull request by glours**
Signed-off-by: Guillaume Lours <705411+glours@users.noreply.github.com>

**What I did**
Bump version of Docker binaries
To limit the impact of this PR, I didn't change the `timeout` type in the API and just add a conversion function for now

**(not mandatory) A picture of a cute animal, if possible in relation with what you did**
image

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -143,11 +147,10 @@ require (
...
 )
 
-require github.com/gofrs/uuid v4.2.0+incompatible // indirect
-
 replace (
-	github.com/docker/cli => github.com/docker/cli v20.10.3-0.20220309205733-2b52f62e9627+incompatible
-	github.com/docker/docker => github.com/docker/docker v20.10.3-0.20220309172631-83b51522df43+incompatible
+	github.com/docker/cli => github.com/docker/cli v20.10.3-0.20221013132413-1d6c6e2367e2+incompatible // 22.06 master branch
+	github.com/docker/docker => github.com/docker/docker v20.10.3-0.20221013203545-33ab36d6b304+incompatible // 22.06 branch
+	github.com/moby/buildkit => github.com/moby/buildkit v0.10.1-0.20220816171719-55ba9d14360a // same as buildx
```
Comment (thaJeztah) [inline go.mod:154]: oh! compose is already using 22.06 branch? Hm...

Comment (glours) [reply go.mod:154]: yes, it was needed to use latest features offered by buildx (which is doing the same btw)

File: pkg/compose/build_buildkit.go [hunk 2]
```diff
@@ -95,16 +95,16 @@ func (s *composeService) getDrivers(ctx context.Context) ([]build.DriverInfo, er
...
 			if fac.Name() == ng.Driver {
 				f = fac
 				continue
 			}
 		}
 		if f == nil {
-			if f = driver.GetFactory(ng.Driver, true); f == nil {
-				return nil, fmt.Errorf("failed to find buildx driver %q", ng.Driver)
+			if f, err = driver.GetFactory(ng.Driver, true); f == nil || err != nil {
+				return nil, fmt.Errorf("failed to find buildx driver %q, error: %s", ng.Driver, err.Error())
```
Comment (milas) [inline pkg/compose/build_buildkit.go:107]:
```suggestion
			if f, err = driver.GetFactory(ng.Driver, true); f == nil || err != nil {
				return nil, fmt.Errorf("failed to find buildx driver %q, error: %v", ng.Driver, err)
```

Minor: I assume it's an invariant that `f != nil` if `err == nil` so this is probably excessively defensive

Comment (nicksieger) [inline pkg/compose/build_buildkit.go:107]: minor:
```suggestion
				return nil, fmt.Errorf("failed to find buildx driver %q, error: %w", ng.Driver, err)
```

**Review by nicksieger — APPROVED** [review]
LGTM

**Review by milas — APPROVED** [review]

File: pkg/compose/build_buildkit.go (hunk 2 above)
Comment (milas) [reply pkg/compose/build_buildkit.go:107]: lol you beat me by 3 mins!

**Review by laurazard — APPROVED** [review]

**Pull request merged**